deltachat-ios/Chat/ChatViewController.swift

```diff
@@ -158,22 +158,55 @@ class ChatViewController: UITableViewController, UITableViewDropDelegate {
         return dcContext.getChat(chatId: chatId)
     }()
 
-    private lazy var contextMenu: ContextMenuProvider = {
-        let config = ContextMenuProvider()
+    private func reactionsMenu(indexPath: IndexPath) -> ContextMenuProvider.ContextMenuItem {
+
+        let messageId = messageIds[indexPath.row]
+        let myReactions = dcContext.getMessageReactions(messageId: messageId)?.reactions.filter { $0.isFromSelf } .map { $0.emoji } ?? []
+
+        let reactionsMenuItems = DefaultReactions.allCases.map { reaction in
+            let sentThisReaction = myReactions.contains(where: { $0 == reaction.emoji })
+            let checkmarkImageName: String?
+            if sentThisReaction {
+                checkmarkImageName = "checkmark"
+            } else {
+                checkmarkImageName = nil
+            }
+
+            return ContextMenuProvider.ContextMenuItem(title: reaction.emoji, imageName: checkmarkImageName) { [weak self] indexPath in
+                guard let self else { return }
+
+                let messageId = self.messageIds[indexPath.row]
+                if sentThisReaction {
+                    dcContext.sendReaction(messageId: messageId, reaction: nil)
+                } else {
+                    dcContext.sendReaction(messageId: messageId, reaction: reaction.emoji)
+                }
+            }
+        }
+
+        let reactionsItem = ContextMenuProvider.ContextMenuItem(
+            title: String.localized("react"),
+            imageName: "heart.circle.fill",
```

Comment:
this is not visible to me, i'd also prefer `face.smiling` as this better fits to the desktop UI using a similar icon for "React"

Reply:
Oh, yes, you're right, good point!
Turns out: If you don't use an image in a menu element, there's no image visible.

Thank you!